### ISSUE TITLE
Accessibility : Site Creation UI Improvements

### ISSUE THREAD

**Issue by jd-alexander** [wordpress-mobile/WordPress-Android#10897]
For this audit the governing principles are as follows ([inspired by a previous audit](https://github.com/wordpress-mobile/WordPress-Android/issues/7288))

- All the clickable elements should be marked as so.
- All nonactionable and decorative elements should be non-focusable.
- Relevant content should be grouped together.
- All important elements should be described through content description.

The following is a list of opportunities to improve the accessibility of the Editor. 

### Site Creation - WordPress.com sites

* Site Creation Step 1
	* Text contrast
	org.wordpress.android:id/input
	The item's text contrast ratio is 3.16. This ratio is based on an estimated foreground color of #8E9196 and an estimated background color of #FFFFFF. Consider using a contrast ratio greater than 4.50 for small text, or 3.00 for large text.
	
* Site Creation Step 2
	* Text contrast
	org.wordpress.android:id/site_info_tag_line
	The item's text contrast ratio is 2.95. This ratio is based on an estimated foreground color of #8E9196 and an estimated background color of #F6F7F7. Consider increasing this item's text contrast ratio to 3.00 or greater.
	
* Site Creation Step 3
	* Touch target
	org.wordpress.android:id/container
	This item's height is 1dp. Consider making the height of this touch target 48dp or larger.

* Site Creation Step 3 - Site Selected
	* Text contrast
	org.wordpress.android:id/domain_suggestion
	The item's text contrast ratio is 1.74. This ratio is based on an estimated foreground color of #C3C4C7 and an estimated background color of #FFFFFF. Consider increasing this item's text contrast ratio to 3.00 or greater.
	
	* Touch target - (Ignore this when shown in Accessibility Scanner) 
	org.wordpress.android:id/container
	This item's height is 17dp. Consider making the height of this touch target 48dp or larger.
	
	

### Site Creation - self hosted site Step 1

* Touch target
org.wordpress.android:id/input
This item's height is 43dp. Consider making the height of this touch target 48dp or larger.

* Text contrast
org.wordpress.android:id/input
The item's text contrast ratio is 3.17. This ratio is based on an estimated foreground color of #3895BA and an estimated background color of #F6F7F7. Consider using a contrast ratio greater than 4.50 for small text, or 3.00 for large text.

**Comment by oguzkocer** [comment]
This issue is targeting the current frozen milestone which is about to be released. I am bumping it to `14.5` which is when a PR that lands to `develop` in the next 2 weeks will be released. If it's more urgent than that, please make sure to target `release/14.4` branch which will be available later today and update the milestone for the issue accordingly.

**Comment by antonis** [comment]
Closing this issue since the site creation flow has changed from the time the issue was created
